ros2_generate_interface_docs/ros2_generate_interface_docs/main.py

```diff
@@ -14,11 +14,79 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import argparse
+import os
 import sys
+import time
+
+from ros2_generate_interface_docs import msg_utils
+from ros2_generate_interface_docs import utils
+
+from rosidl_runtime_py import get_message_interfaces
+
+
+def generate_interfaces(interfaces, html_dir, template, interface_type):
+    """
+    Generate the index and documentation for each message.
+
+    :param interfaces: dictionary with the interface package name associated with all
+        the interface for this package.
+    :type interfaces: dict of {str : str[]}
+    :param html_dir: path to the directory to save the generated documentation
+    :type html_dir: str
+    :param template_dir: template name
+    :type template_dir: str
+    :param interface_type: type of the interface: msg, action or srv
+    :type interface_type: str
+    """
+    timestamp = str(time.strftime('%a, %d %b %Y %H:%M:%S'))
```

Comment:
A common pattern is to separate the model from the view. In this case the Python logic should determine the current time and pass it to the template. The template then formats its with a specific format string.